src/nft/components/bag/ButtonStates.tsx

```diff
@@ -0,0 +1,148 @@
+import { Trans } from '@lingui/macro'
+import { PriceImpact } from 'nft/hooks/usePriceImpact'
+import { ReactNode } from 'react'
+import { DefaultTheme } from 'styled-components/macro'
+
+export enum BuyButtonStates {
+  WALLET_NOT_CONNECTED,
+  NOT_SUPPORTED_CHAIN,
+  INSUFFICIENT_BALANCE,
+  ERROR,
+  IN_WALLET_CONFIRMATION,
+  PROCESSING_TRANSACTION,
+  FETCHING_TOKEN_ROUTE,
+  INVALID_TOKEN_ROUTE,
+  NO_TOKEN_ROUTE_FOUND,
+  LOADING_ALLOWANCE,
+  IN_WALLET_ALLOWANCE_APPROVAL,
+  PROCESSING_APPROVAL,
+  REQUIRE_APPROVAL,
+  CONFIRM_UPDATED_PRICE,
+  PRICE_IMPACT_HIGH,
+  PAY,
+}
+
+export interface BuyButtonState {
+  handleClick: (() => void) | (() => Promise<void>)
+  buttonText: ReactNode
+  disabled: boolean
+  warningText: ReactNode | undefined
+  warningTextColor: string
+  helperText: ReactNode | undefined
+  helperTextColor: string
+  buttonColor: string
+  buttonTextColor: string
+}
+
+export function getBuyButtonStateData(
```

Comment:
nit: If you rename this to `useBuyButtonStateData` to treat it as a hook, you can then use `const theme = useTheme` to not have to pass the theme object with each call.